components/com_content/src/Model/CategoryFormModel.php

```diff
@@ -0,0 +1,501 @@
+use Joomla\Component\Categories\Administrator\Table\CategoryTable;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * Frontend category form model for com_content.
+ *
+ * @since  6.1.0
+ */
+class CategoryFormModel extends AdminCategoryModel
+{
+    /**
+     * Component option name.
+     *
+     * @var    string
+     * @since  6.1.0
+     */
+    protected $option = 'com_content';
+
+    /**
+     * Model typeAlias string. Used for version history.
+     *
+     * @var    string
+     * @since  6.1.0
+     */
+    public $typeAlias = 'com_content.category';
+
+    /**
+     * Override getState to short-circuit heavy populate logic and set the ID explicitly.
+     *
+     * @param   string  $property  Optional parameter name.
+     * @param   mixed   $default   Optional default value.
+     *
+     * @return  mixed
+     *
+     * @since   6.1.0
+     */
+    public function getState($property = null, $default = null)
+    {
+        if ($this->state === null) {
+            $this->state = new Registry();
+        }
+
+        if (!$this->__state_set) {
+            $app       = Factory::getApplication();
+            $input     = $app->getInput();
+            $id        = $input->getInt('id');
+            $parentId  = $input->getInt('parent_id');
+            $extension = $input->getCmd('extension', 'com_content');
+            $parts     = explode('.', $extension);
+
+            $this->state->set($this->getName() . '.id', $id);
+            $this->state->set('categoryform.id', $id);
+            $this->state->set('category.id', $id);
+            $this->state->set('category.parent_id', $parentId);
+            $this->state->set('category.extension', $extension);
+            $this->state->set('category.component', $parts[0]);
+            $this->state->set('category.section', $parts[1] ?? null);
+            $this->state->set('params', $app->getParams());
+            $this->state->set('return_page', base64_decode($input->get('return', '', 'base64')));
+            $this->state->set('layout', $input->getCmd('layout', 'edit'));
+
+            if (!$id && Multilanguage::isEnabled()) {
+                $this->state->set('category.language', Factory::getLanguage()->getTag());
+            }
+
+            $this->__state_set = true;
+        }
+
+        return $property === null ? $this->state : $this->state->get($property, $default);
+    }
+
+    /**
+     * Method to auto-populate the model state.
+     *
+     * @return  void
+     *
+     * @since   6.1.0
+     */
+    protected function populateState()
+    {
+        // State is set in getState override to avoid recursion during debugging.
+    }
+
+    /**
+     * Method to get category data.
+     *
+     * @param   integer  $pk  The id of the category.
+     *
+     * @return  mixed  Category item data object on success, false on failure.
+     *
+     * @since   6.1.0
```

Comment:
```suggestion
     * @since   __DEPLOY_VERSION__
```

doing this means our build scripts will correctly set the version - it might not make it to 6.1 ;)